tensorflow/java/src/main/java/org/tensorflow/EagerSession.java

```diff
@@ -208,27 +195,130 @@ private Options() {
+   * @see {@link #getDefault()}
+   */
+  public static EagerSession initDefault(Options options) {
+    synchronized(EagerSession.class) {
+      if (defaultSession != null) {
+        throw new IllegalStateException("Default eager session is already initialized");
+      }
+      defaultSession = options.build();
+    }
+    return defaultSession;
+  }
+
+  /**
+   * Returns the default eager session
+   * 
+   * <p>Once initialized, the default eager session remains active for the whole life of the 
+   * application, as opposed to sessions obtained from {@link #create()} or {@link Options#build()} 
+   * which should be closed after their usage.
+   * 
+   * <p>The default set of {@link Options} is used to initialize the session on the first call. To 
+   * override this behavior, it is possible to invoke {@link #initDefault(Options)} with a different 
+   * set of options prior to this first call. 
+   * 
+   * <p>Example usage:
+   * 
+   * <pre>{@code
+   * // Starting to build eager operations using default session, by calling 
+   * // EagerSession.getDefault() implicitly
+   * Ops tf = Ops.create();
+   * 
+   * // Starting to build eager operations using default session, by calling 
+   * // EagerSession.getDefault() explictly
+   * Ops tf = Ops.create(EagerSession.getDefault());
+   * }</pre>
+   * 
+   * @return default eager session
+   * @see {@link #initDefault(Options)}
+   */
+  public static EagerSession getDefault() {
+    if (defaultSession == null) {
+      synchronized(EagerSession.class) {
+        if (defaultSession == null) {
+          defaultSession = options().build();
+        }
+      }
+    }
+    return defaultSession;
   }
 
-  /** Returns an {@code EagerSession} configured with default options. */
+  /** 
+   * Returns an {@code EagerSession} configured with default options. 
+   *
+   * <p><b>WARNING:</b>Instances of {@code EagerSession} returned by this method must be explicitly freed
+   * by invoking {@link #close()} when they are no longer needed. This could be achieve using
+   * the `try-with-resources` technique.
+   * 
+   * <p>Example usage:
+   *
+   * <pre>{@code
+   * try (EagerSession session = EagerSession.create()) {
+   *   Ops tf = Ops.create(session);
+   *   // build execute operations eagerly...
+   * }
+   * }</pre>
+   */
   public static EagerSession create() {
     return options().build();
   }
 
-  private EagerSession(Options options) {
-    this.nativeHandle = allocate(options.async, options.devicePlacementPolicy.code, options.config);
-    this.resourceCleanupStrategy = options.resourceCleanupStrategy;
-
-    if (resourceCleanupStrategy == ResourceCleanupStrategy.IN_BACKGROUND) {
-      nativeResources.startCleanupThread();
-    }
+  /** 
+   * Returns an object that configures and builds a {@code EagerSession} with custom options. 
+   * 
+   * <p><b>WARNING:</b>Instances of {@code EagerSession} returned by this method must be explicitly freed
```

Comment:
is this warning supposed to be attached to the `options` method? Just checking if it's accidentally copy-pasted from the method above.

Reply:
Mmmh, good point... I intend to put the warning there but it could probably be rephrased as

> Instances of {@code EagerSession} returned by {@link EagerSession.Options.build()} must be explicitly freed...

Or it could be moved to `EagerSession.Options.build()`... what do you think?